cli/compile/compile.go

```diff
@@ -57,12 +57,16 @@ var (
 // NewCommand created a new `compile` command
 func NewCommand() *cobra.Command {
 	command := &cobra.Command{
-		Use:     "compile",
-		Short:   "Compiles Arduino sketches.",
-		Long:    "Compiles Arduino sketches.",
-		Example: "  " + os.Args[0] + " compile -b arduino:avr:uno /home/user/Arduino/MySketch",
-		Args:    cobra.MaximumNArgs(1),
-		Run:     run,
+		Use:   "compile",
+		Short: "Compiles Arduino sketches.",
+		Long:  "Compiles Arduino sketches.",
+		Example: "" +
+			"  " + os.Args[0] + " compile -b arduino:avr:uno /home/user/Arduino/MySketch\n" +
+			"  " + os.Args[0] + " compile -b arduino:avr:uno --build-property=\"build.extra_flags=\\\"-DMY_DEFINE=\\\"hello world\\\"\\\"\" /home/user/Arduino/MySketch\n" +
+			"  " + os.Args[0] + " compile -b arduino:avr:uno --build-property=\"build.extra_flags=-DPIN=2 \\\"-DMY_DEFINE=\\\"hello world\\\"\\\"\" /home/user/Arduino/MySketch\n" +
+			"  " + os.Args[0] + " compile -b arduino:avr:uno --build-property=build.extra_flags=-DPIN=2 --build-property=\"compiler.cpp.extra_flags=\\\"-DSSID=\\\"hello world\\\"\\\"\"-DMY_DEFINE=\"hello world\"' /home/user/Arduino/MySketch\n",
```

Comment:
are you sure that this last example is correct? on the help is displayed as:

```
  ./arduino-cli compile -b arduino:avr:uno --build-property=build.extra_flags=-DPIN=2 --build-property="compiler.cpp.extra_flags=\"-DSSID=\"hello world\"\""-DMY_DEFINE="hello world"' /home/user/Arduino/MySketch
```

and it doesn't look right to me...
To simplify your life you may use the golang verbatim string operator (backtick) so the help strings will not require triple quoting:

```
  `--build-property="build.extra_flags=\"-DMY_DEFINE=\"hello world\"\"" /home/user/Arduino/MySketch`
```

another small improvement may be to drop the (optional) `=` sign after `--build-property` so instead of:

  `--build-property="build.extra_flags=\"-DMY_DEFINE=\"hello world\"\""`

we may write

  `--build-property "build.extra_flags=\"-DMY_DEFINE=\"hello world\"\""`

that is slightly clearer IMHO.

Reply:
Right! The last example is plainly wrong, I'll fix it.

Somehow I always forget about the backtick. :facepalm: